corehq/ex-submodules/phonelog/reports.py

```diff
@@ -153,20 +153,18 @@ def breadcrumbs(self):
     def rendered_report_title(self):
         new_title = self.name
         if self.errors_only:
-            new_title = (
-                "Errors &amp; Warnings Log <small>for %s</small>" % (
-                    ", ".join(self.device_log_users)
-                )
-                if self.device_log_users
-                else "Errors &amp; Warnings Log"
-            )
+            new_title = format_html(
+                "Errors &amp; Warnings Log <small>for {}</small>",
+                ", ".join(self.device_log_users)
+            ) if self.device_log_users else "Errors & Warnings Log"
         elif self.goto_key:
             log = self.goto_log
-            new_title = "Last %s Logs <small>before %s</small>" % (
+            new_title = format_html(
+                "Last {} Logs <small>before {}</small>",
                 self.limit,
                 ServerTime(log.date).user_time(self.timezone).ui_string()
             )
-        return mark_safe(new_title)
+        return new_title
```

Comment:
why not just change the call here instead of the two places where the string is being made? Does it ensure something?

Reply:
Great question. IMO, pushing the safety burden as low as possible decreases the chances someone makes a mistake. It also makes the code more flexible. Here, it's just two conditions, both of which call `format_html`, but if it were 3 conditions, where one of them only returned user input, for example, then just using a blanket `mark_safe` at the bottom wouldn't work. Non-theoretically, it is also the most practical way to do as [the docs](https://docs.djangoproject.com/en/3.1/ref/utils/#django.utils.html.format_html) recommend: use `format_html` when building up HTML fragments.

Reply:
Sweet, thanks.

Reply:
that makes sense, thank you!